pkg/mysqlcluster/orc_reconciliation.go

```diff
@@ -72,8 +74,96 @@ func (f *cFactory) SyncOrchestratorStatus(ctx context.Context) error {
 	return nil
 }
 
+func getInstance(hostname string, insts []orc.Instance) (orc.Instance, error) {
+	for _, node := range insts {
+		host := node.Key.Hostname
+		if host == hostname {
+			return node, nil
+		}
+	}
+	return insts[0], fmt.Errorf("the element was not found")
+}
+
+func getMaster(node orc.Instance, insts []orc.Instance) (string, error) {
+
+	if len(node.MasterKey.Hostname) != 0 && node.IsCoMaster == false {
+		next, err := getInstance(node.MasterKey.Hostname, insts)
+		if err == nil {
+			return getMaster(next, insts)
+		} else {
+			return "", err
+		}
+	}
+
+	if node.IsCoMaster == true {
+		next, err := getInstance(node.MasterKey.Hostname, insts)
+		if err == nil {
+			return next.Key.Hostname, nil
+		} else {
+			return "", err
+		}
+	}
+
+	return node.Key.Hostname, nil
+}
+
+func verifyMasterArray(masterForNode []string) (string, bool) {
+	masterHostName := masterForNode[0]
+	var check bool = true
+	for _, node := range masterForNode {
+		if node != masterHostName {
+			check = false
+		}
+
+	}
+
+	return masterHostName, check
+}
+
+func (f *cFactory) setMasterWritable(masterHostName string, insts []orc.Instance) {
+	master, err := getInstance(masterHostName, insts)
```

Comment:
you can compute `masterHostName` using `determinMasterFor` and don't need to pass it as parameter. See see the next comments.